### PR TITLE
Feature/specification

### PR DESCRIPTION
**Pull request by cvarjao**
Tasks:
- [ ] Port assignment/reservation (e.g.: Tomcat)
- [ ] deploymentSpec vs runtimeSpec ... do we need them?
- [ ] requires format (e.: ojdbc.url, ojdbc.oid.url). multiple formats?
- [ ] support selector equivalent of kubernetes `matchLabels` or `matchExpressions`

### CODE REVIEW

**Comment by cvarjao** [comment]
@GeorgeWalker, the wiki seems to be a bit limited on the collaboration/back-and-forth.
I am hoping a PR will allow us to have a place to review, discuss, and collaborate.

PS.: I am doing the edits online, so it is a lot of commits. at some point I will squash them in 1 commit.

File: spec/example-artifacts.json [hunk 1]
```diff
@@ -0,0 +1,110 @@
+[
+  {
+    "system": "OFM",
+    "symbolicName": "com.oracle.ofm",
```
Comment (brendenblack) [inline spec/example-artifacts.json]: Is this field meant to match Maven's group ID?

Comment (cvarjao) [reply spec/example-artifacts.json]: yes. That should actually be "name". I will update.

**Comment by GeorgeWalker** [comment]
Thanks for the updated examples.

**Pull request merged**